share/mk/src.opts.mk

```diff
@@ -372,7 +372,9 @@ BROKEN_OPTIONS+=GOOGLETEST SSP
 BROKEN_OPTIONS+=NS_CACHING
 .endif
 
-.if ${__C} == "cheri" || ${__T:Mmips64*c*} || ${__T:Mriscv*c*} || ${.MAKE.OS} == "Linux"
+.if ${__C} == "cheri" || ${__C} == "morello" || \
+    ${__T} == "aarch64c" || ${__T:Mmips64*c*} || \
+    ${__T:Mriscv*c*} || ${.MAKE.OS} == "Linux"
```

Comment:
Given this affects the kernel and this PR logically depends on kernel support I am moving this change to #842.